staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -98,17 +102,21 @@ func WithMaxInFlightLimit(
 				handler.ServeHTTP(w, r)
 			default:
 				metrics.MonitorRequest(r, strings.ToUpper(requestInfo.Verb), requestInfo.Resource, requestInfo.Subresource, "", errors.StatusTooManyRequests, time.Now())
-				tooManyRequests(r, w)
+				tooManyRequests(r, w, disconnectChanceOnMaxRequests)
 			}
 		}
 	})
 }
 
-func tooManyRequests(req *http.Request, w http.ResponseWriter) {
+func tooManyRequests(req *http.Request, w http.ResponseWriter, disconnectChanceOnMaxRequests float64) {
 	// "Too Many Requests" response is returned before logger is setup for the request.
 	// So we need to explicitly log it here.
 	defer httplog.NewLogged(req, &w).Log()
 
+	if disconnectChanceOnMaxRequests > 0 && rand.Float64() < disconnectChanceOnMaxRequests {
+		w.Header().Set("Connection", "close")
```

Comment:
I agree with @liggitt - if we want to force rebalncing, we can't really rely on the client to do the right thing.